### PR TITLE
Apb 1651

### PR DESCRIPTION
**Pull request by habeeb01**

### CODE REVIEW

File: app/uk/gov/hmrc/dynamicconfig/controllers/DynamicConfig.scala [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+case class ErrorBody(code: String, message: String)
+
+class Controller @Inject()(repository: Repository) extends BaseController {
+
+	implicit val propertyFormat 	= Json.format[Property]
+	implicit val valueFormat 			= Json.format[Value]
+	implicit val errorBodyFormat 	= Json.format[ErrorBody]
+
+	def createProperty = Action.async(parse.json) { implicit request =>
+    withJsonBody[Property] { property =>
```
Comment (opetch) [inline app/uk/gov/hmrc/dynamicconfig/controllers/DynamicConfig.scala]: Formatting is wrong, in fact tabs are used in this file (except here) and spaces in the test. IntelliJ picks up the indentation style from the file (if it's set, and is by default). Maybe your IDE is defaulting to tabs on new files? (and maybe that's default). Anyway following advice in the comments here https://intellij-support.jetbrains.com/hc/en-us/community/posts/207045165-Show-non-printable-characters-Tabs- will at least give a visual indication that indentation is wrong.

File: conf/app.routes [hunk 2]
```diff
@@ -1,3 +1,6 @@
 # microservice specific routes
 
-GET        /hello-world             uk.gov.hmrc.dynamicconfig.controllers.MicroserviceHelloWorld.hello
+GET        /property/:key         @uk.gov.hmrc.dynamicconfig.controllers.Controller.getProperty(key: String)
```
Comment (arturopala) [inline conf/app.routes]: it should be in plural: **/properties**

**Comment by arturopala** [comment]
Generally looks fine, but I think it would be better to start with Agents modern template https://github.com/hmrc/agents-service-template.g8 which is closer to Play 2.5 DI standards.

**Comment by arturopala** [comment]
if not, then we have to follow https://confluence.tools.tax.service.gov.uk/pages/viewpage.action?spaceKey=PLATOPS&title=How+to+remove+Play+GlobalSettings to not start service with technical debt.

**Comment by opetch** [comment]
@arturopala hey so we ended up moving our code (routes, controller, repository, tests) into the g8 template you mentioned, getting it working and then stripping out everything that we deemed wasn't required. I've pushed that to master given the non boiler plate code had already been reviewed. However if you could take a look at the project to make sure we didn't do anything silly with the boiler plate, that would be great. Thanks.

**Comment by opetch** [comment]
@arturopala as you suggested I've moved to the newer version of the template this is now in master, when you have a once over a quick check over the boiler plate would be appreciated :+1:

**Comment by arturopala** [comment]
I had a look and it looks good now.

**Pull request closed without merging**